pkg/armhelpers/utils/util.go

```diff
@@ -91,8 +96,14 @@ func K8sLinuxVMNameParts(vmName string) (poolIdentifier, nameSuffix string, agen
 }
 
 // VmssNameParts returns parts of Linux VM name e.g: k8s-agentpool1-11290731-0
+// For windows, the VMSS name is 'aks' concatenated with agentpool name
 func VmssNameParts(vmssName string) (poolIdentifier, nameSuffix string, err error) {
-	vmssNameParts := vmssnameRegexp.FindStringSubmatch(vmssName)
+	vmssNameParts := windowsVmssRegexV2.FindStringSubmatch(vmssName)
+	if len(vmssNameParts) == 2 {
```

Comment:
Do we have Windows/Linux context in the code areas that call this func? Determining Windows vs Linux this way seems edge case-prone. It might be better to pass in to this func whether or not to use the Windows or Linux regex, or to have separate funcs for each.

Reply:
added a new isWindows flag for vmss scale sets to upgrade.